controllers/numaresourcesoperator_controller.go

```diff
@@ -581,6 +581,31 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
 			return nil, fmt.Errorf("failed to apply (%s) %s/%s: %w", objState.Desired.GetObjectKind().GroupVersionKind(), objState.Desired.GetNamespace(), objState.Desired.GetName(), err)
 		}
 	}
+
+	for _, obj := range r.RTEMetricsManifests.ToObjects() {
+		// Check if the object already exists
+		existingObj := obj.DeepCopyObject().(client.Object)
```

Comment:
why do we need the cast here?

Reply:
If I'll remove the casting:
I'll get this error when using `r.Client.Get`:
`cannot use existingObj (variable of type runtime.Object) as client.Object value in argument to r.Client.Get: runtime.Object does not implement client.Object (missing method GetAnnotations)`

Reply:
and why do we use `DeepCopyObject()` and not `DeepCopy()` ?

Reply:
@ffromani This because the client.Object implements DeepCopyObject() and not  DeepCopy()